kroxylicious-filters/kroxylicious-oauthbearer-validation/src/test/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationTest.java

```diff
@@ -132,8 +143,55 @@ void mustCloseOauthHandler() throws Exception {
         verify(callbackHandler).close();
     }
 
+    @Test
+    void initShouldAddJwksEndpointToOauthAllowList() {
+        // given
+        OauthBearerValidation oauthBearerValidation = new OauthBearerValidation(callbackHandler);
+        Config config = defaultConfig(URI.create("https://" + UUID.randomUUID() + ".invalid"));
+
+        // when
+        oauthBearerValidation.initialize(ffc, config);
+
+        // then
+        assertThat(System.getProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG))
+                .isNotEmpty()
+                .contains(config.jwksEndpointUrl().toString());
+    }
+
+    static Stream<Arguments> closeShouldRestoreOauthAllowList() {
+        return Stream.of(
+                Arguments.argumentSet("clears system property after close",
+                        (Consumer<String>) unused -> System.clearProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG),
+                        (BiConsumer<String, String>) (initial, propValue) -> assertThat(propValue).isNull()),
+                Arguments.argumentSet("preserves existing system property value",
+                        (Consumer<String>) initial -> System.setProperty(ALLOWED_SASL_OAUTHBEARER_URLS_CONFIG, initial),
```

Comment:
`initial` makes me think of the System prop before initialization. Maybe this should be called `configValue` in the lambdas?